src/cswinrt/code_writers.h

```diff
@@ -9683,30 +9683,45 @@ namespace WinRT
 {
 % static partial class Module
 {
-public static unsafe IntPtr GetActivationFactory(String runtimeClassId)
-{%
-return IntPtr.Zero;
+public static unsafe IntPtr GetActivationFactory(% runtimeClassId)
+{
+switch (runtimeClassId)
+{
+%
+default:
+    return IntPtr.Zero;
 }
 }
+%
+}
 }
 )",
     internal_accessibility(),
+    settings.net7_0_or_greater ? "ReadOnlySpan<char>" : "string",
```

Comment:
Consider a comment here, that doing a switch on `ReadOnlySpan<char>` it supported on net7 or greater which is what we are taking advantage of.

Reply:
Done in https://github.com/microsoft/CsWinRT/pull/1401/commits/35edf3b2704af993ec6727d8ce6482d5827a8518 🙂